SYCL/AtomicRef/atomic_memory_order_acq_rel.cpp

```diff
@@ -1,33 +1,239 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -O3 -o %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_70
 // RUN: %HOST_RUN_PLACEHOLDER %t.out
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
 // RUN: %ACC_RUN_PLACEHOLDER %t.out
 // L0, OpenCL, and HIP backends don't currently support
 // info::device::atomic_memory_order_capabilities
-// UNSUPPORTED: level_zero || opencl || hip
+// UNSUPPORTED: level_zero, opencl, hip
 
-// NOTE: Tests load and store for acquire-release memory ordering.
+// host does not support barrier
+// XFAIL: host
+
+// NOTE: Tests fetch_add for acquire and release memory ordering.
 
 #include "atomic_memory_order.h"
 #include <iostream>
+#include <numeric>
 using namespace sycl;
 
-int main() {
+template <memory_order order> void test_acquire_global() {
+  const size_t N_items = 1024;
+  const size_t N_iters = 100000;
```

Comment:
I am fine with reducing the number of iterations. What number would you like to see? Also let me just point out that the test is still relatively fast, despite seemingly high number of iterations.

Reply:
I would probably just cut a 0, but if it isn't that bad w.r.t execution time then maybe its okay. How long is the execution time in comparison to AtomicRef/add.cpp?

Reply:
Maybe if we somehow reduce the number of kernels submitted, that would help? I.e. if we factor out kernel body instead of kernel submission function.

Reply:
The kernels are actually different. I tried grouping all the tests for each memory order into one kernel, but it makes the code a bit less readable, while reducing the test runtime by 1.5s. I am not sure if that is worth it.

Reply:
@steffenlarsen what is your preference here - how much should we adapt the code for minimal running time?

Reply:
I'm thinking maybe it's best if we make it do 1000 iterations and live with whatever cost that has. I understand that 1000 iterations isn't necessarily a lot when testing for something like this, but hopefully with the amount that it will be run we will eventually find bugs here. Are you okay with that approach?

Reply:
@steffenlarsen which direction would you like me to take with this? Keep it as it is or merge multiple kernels?

Reply:
Oh, apologies. I honestly thought I had responded. I am okay with keeping them as multiple kernels for the sake of readability.

Reply:
I reduced the number of operations to 1000.